python/lsst/ts/ledprojector/ledprojector_csc.py

```diff
@@ -0,0 +1,256 @@
+        initial_state: salobj.State = salobj.State.STANDBY,
+        override: str = "",
+        simulation_mode: bool = False,
+    ) -> None:
+        self.led_controller = None
+
+        self.config = None
+
+        super().__init__(
+            name="LEDProjector",
+            index=0,
+            config_schema=CONFIG_SCHEMA,
+            config_dir=config_dir,
+            initial_state=initial_state,
+            override=override,
+            simulation_mode=simulation_mode,
+        )
+
+    @staticmethod
+    def get_config_pkg() -> str:
+        return "ts_config_mtcalsys"
+
+    @property
+    def controller_connected(self) -> bool:
+        """Return True if the LabJack is connected"""
+        return self.led_controller is not None and self.led_controller.connected
+
+    async def configure(self, config: Any) -> None:
+        config = copy.copy(config)
+        self.config = config
+
+    # TODO should we be catching errors from LEDController
+    async def connect_led(self) -> None:
+        """Connect to the LabJack and get status.
+        This method initiates the LEDController as well
+
+        Raises
+        ------
+        ValueError
+            If there is no config to configure the LEDController with
+        asyncio.TimeoutError
+            If it takes longer than self.config.led.connect_timeout
+        """
+        await self.disconnect_led()
+        if self.led_controller is None:
+            if self.config is None:
+                raise ValueError(
+                    "CSC tried to use LEDController with config equal to None"
+                )
+            self.led_controller = LEDController(
+                config=self.config,
+                log=self.log,
+                simulate=self.simulation_mode,
+            )
+        await asyncio.wait_for(
+            self.led_controller.connect(),
+            timeout=self.led_controller.COMMUNICATION_TIMEOUT,
+        )
+
+    # TODO should we be catching errors
+    async def disconnect_led(self) -> None:
+        """Disconnect to the LabJack & delete the LEDController object
+
+        Raises
+        ------
+        Exception
+            If disconnect failed
+        """
+        try:
+            if self.led_controller is None:
+                return
+            await self.led_controller.disconnect()
+            # Delete the led controller because the config may change.
+            self.led_controller = None
+        except Exception as e:
+            self.log.warning(f"Failed to disconnect led; continuing: {e!r}")
+
+    async def start(self) -> None:
+        await super().start()
+
+    async def close_tasks(self) -> None:
+        """Close the CSC gratefully.
+
+        Disconnects the labjack, deletes LEDController object
+        Then closes tasks.
+
+        """
+        await self.disconnect_led()
+        await super().close_tasks()
+
+    # TODO should I just try to connect first if not connected
```

Comment:
No, I don't think so. However, you should probably have a background task that is checking the CSC is connected when it is in DISABLED/ENABLED state and send the CSC to fault if it ever looses connection.